### PR TITLE
commands/envelope: Fall back to account for signing key

### PR DESCRIPTION
**Pull request by dcbaker**
If there isn't a key provided as an argument to sign or togglesign, fall
back to using the account of the sending address to determine the key,
otherwise the message will be marked as "to sign", but won't actually be
signed. This is the same type of logic used for sign_by_default.

### CODE REVIEW

**Comment by pazz** [comment]
I like it. On a side node regarding the new `else` clause:
Maybe we should add a dedicated methon in `Envelope` or a related utility class that extracts a suitable accound from an envelope's From-header. The following code is already replicated in `alot.commands.envelope`:

```python
_, saddr = email.Utils.parseaddr(msg.get('From', ''))
account = settings.get_account_by_address(saddr)
if account is None:
    if not settings.get_accounts():
        ui.notify('no accounts set', priority='error')
        return
    else:
        account = settings.get_accounts()[0]
```

Since `db.envelope.Envelope` already refers to the settings manager internally, this could simply be a new Envelope method I believe.

**Comment by dcbaker** [comment]
I think the helper is a good idea. I put it in utils instead of the Envelope class, I just couldn't justify why it should be in that class other than that it would be convenient, so I made it a utils helper. It's part of this series now.

**Comment by dcbaker** [comment]
bump

File: alot/commands/utils.py [hunk 1]
```diff
@@ -90,3 +96,27 @@ def _get_keys(ui, encrypt_keyids, block_error=False, signed_only=False):
                 continue
         keys[crypto.hash_key(key)] = key
     returnValue(keys)
+
+
+def get_account_by_address(addr, return_default=False):
+    """Get an account given and address.
```
Comment (pazz) [inline alot/commands/utils.py]: type *an ?

File: alot/commands/utils.py [hunk 2]
```diff
@@ -90,3 +96,27 @@ def _get_keys(ui, encrypt_keyids, block_error=False, signed_only=False):
...
+
+    :param str addr: The address to get an account for.
+    :param bool return_default: If true returns the first account if no account
+        can be found matching the addr.
+    :raises `~alot.errors.NoMatchingAccount`: If return_default if false and no
```
Comment (pazz) [inline alot/commands/utils.py]: typo `s/if/is/`

**Comment by dcbaker** [comment]
I've extended the Manager method with the return_default keyword and updated tests and documentation accordingly.

**Comment by dcbaker** [comment]
thanks! merged

**Pull request merged**